### ISSUE TITLE
Package version differences outside vs inside

### ISSUE THREAD

**Issue by roddtalebi** [ezCGP/ezCGP#274]
Here are the packages we have access to in the room:

- python 3.7.13
- deap 1.0
- torch 1.3.1
- torchvision 0.4.2
- PIL 9.0.1
- cudatoolkit 10.0

**Comment by roddtalebi** [comment]
a few issues i've already come accross:

- to import torch vision, had to edit the package code to not use PIL.PILLOW_VERSION
- deap.tools.selNSGA2 only takes in individuals, k
- torch.cat can only handle dim if it is int not np.int64 or something
- torchvision had issues when in offline mode

**Comment by roddtalebi** [comment]
```
conda create -n closed python=3.7.13
conda activate closed
conda install cudatoolkit=10.0 pytorch=1.3.1 torchvision=0.4.2 -c pytorch
```
edit <PATH_TO_CONDA_ENVS>/closed/lib/python3.7/site-packages/torchvision/transforms/functional.py to remove dependence on PILLOW_VERSION
```
conda install typing numpy pandas matplotlib scipy six pillow opencv scikit-learn mpi4py
pip install deap==1.0.0
```

**Comment by roddtalebi** [comment]
An issue for after we submit the paper, but any idea what the purpose of this line is?
https://github.com/ezCGP/ezCGP/blob/58e1d781a0abe0acfba5d13b7aa2f9aa92d6b650/codes/utilities/simgan_fid_metric.py#L66

old versions of torchvision (ie the one we have in the closed room) doesn't have this attribute for the inception model

- version in the room https://github.com/pytorch/vision/blob/v0.4.2/torchvision/models/inception.py#L141
- version we use https://github.com/pytorch/vision/blob/release/0.8.0/torchvision/models/inception.py#L110

looks like in the old version they just put it in forward() so there is no way to get info on it by calling the model. I don't really get what register_forward_hook does especially with everything else around it...help!

If we can say what it is supposed to do, then we can find an alternative way to do it hopefully to work with older versions of torchvision. On that note, it looks like they switch out of calling it in forward in 0.7.0 https://github.com/pytorch/vision/blob/v0.7.0/torchvision/models/inception.py#L110

**Comment by roddtalebi** [comment]
I think I got it...

looked at this for help
https://pytorch.org/docs/stable/generated/torch.nn.modules.module.register_module_forward_hook.html

basically a cheat way to get the input and output of a certain module/layer after it's forward call. cute.

going to register the previous layer https://github.com/pytorch/vision/blob/v0.4.2/torchvision/models/inception.py#L138 and then just manually apply the adaptive avg pool method to the output of that hook. i didn't really read into what adaptive avg pool does but i doubt it is random.
